auth/src/main/java/com/firebase/ui/auth/AuthUI.java

```diff
@@ -529,14 +527,94 @@ public Intent build() {
         }
 
         @VisibleForTesting()
+        public abstract FlowParameters getFlowParams();
+    }
+
+    /**
+     * Builder for the intent to start the reauthentication flow.
+     */
+    public final class ReauthIntentBuilder extends AuthIntentBuilder<ReauthIntentBuilder> {
+        private String mReauthReason;
+        private String mReauthEmail;
+
+        private ReauthIntentBuilder() {
+            super();
+        }
+
+        /**
+         * Set an explanation for why reauth was requested e.g. "To delete your account you must
+         * reauthenticate."
+         *
+         * @param reauthReason A string explaining why reauthentication was requested.
+         * @return
+         */
+        public ReauthIntentBuilder setReauthReason(String reauthReason) {
+            mReauthReason = reauthReason;
+            return this;
+        }
+
+        /**
+         * Sets the email address for which reauthentication will be tried.  If email accounts are
+         * enabled this MUST be set or email reauthentication will fail.
+         *
+         * @param reauthEmail The email address of the account for which reauthentication is being
+         * requested
+         */
+        public ReauthIntentBuilder setReauthEmail(String reauthEmail) {
+            mReauthEmail = reauthEmail;
+            return this;
+        }
+
+        @Override
+        public FlowParameters getFlowParams() {
+            return new FlowParameters(
+                    mApp.getName(),
+                    new ArrayList<>(mProviders),
+                    mTheme,
+                    mLogo,
+                    mTosUrl,
+                    mIsSmartLockEnabled,
```

Comment:
@amandle Have you tried reauthenticating with smart lock enabled? Given @samtstern's comment about showing a dialog before we show auth stuff, I feel like things will get complicated with smart lock enabled. @samtstern @amandle Am I missing something or could this be a problem?